ddtrace/llmobs/_integrations/openai.py

```diff
@@ -188,17 +181,19 @@ def _extract_llmobs_metrics_tags(span: Span, resp: Any, span_kind: str) -> Dict[
                 OUTPUT_TOKENS_METRIC_KEY: output_tokens,
                 TOTAL_TOKENS_METRIC_KEY: input_tokens + output_tokens,
             }
-
             # Chat completions returns `prompt_tokens_details` while responses api returns `input_tokens_details`
             prompt_tokens_details = _get_attr(token_usage, "prompt_tokens_details", {}) or _get_attr(
                 token_usage, "input_tokens_details", {}
             )
             cached_tokens = _get_attr(prompt_tokens_details, "cached_tokens", None)
             if cached_tokens is not None:
                 metrics[CACHE_READ_INPUT_TOKENS_METRIC_KEY] = cached_tokens
-
             return metrics
-        return get_llmobs_metrics_tags("openai", span)
+        elif kwargs.get("stream") and resp is not None:
+            return get_token_metrics_from_streamed_response(
+                span, resp, kwargs.get("prompt", None), kwargs.get("messages", None), kwargs
+            )
+        return None
```

Comment:
I like that we've moved out extracting token metrics to the integration level, but I can't help but notice `get_token_metrics_from_streamed_response` has almost 1:1 duplicate code with  https://github.com/DataDog/dd-trace-py/pull/13791/files#diff-5b467cc82f5b96a7124a28da35a07477e10b42bd51b02903f8985854bc14b824R166-R183. Can we modify this to avoid having redundant code? i.e. if we notice that the token usage is not provided (which means it was a streamed response with user explicitly setting `return_token_info=False`), can we just directly run token estimation helpers?